### PR TITLE
Fix #15354, #15315: Uninitialized collapsibles should not `toggle` on `.collapse('hide')`

### LINKED ISSUE #15315: .collapse('hide') on a hidden uninitialized collapsible shows it
Calling `collapse('hide')` on a collapse panel that is collapsed and has not yet been initialized will toggle the panel (which starts the transition to being shown) when it creates the new Collapse object. Then the call to `data[option]()` won't hide the panel because it is  still transitioning.

### LINKED ISSUE #15354: .collapse('hide') as onclick on a radiobutton   SHOWs the content
see:
http://stackoverflow.com/questions/24466260/clicking-on-a-radio-button-explicitly-coded-to-close-all-bootstrap-accordions-op

http://jsfiddle.net/3F6TY/3/

### PR DESCRIPTION
**Pull request by nikita240**
Fix https://github.com/twbs/bootstrap/issues/15315
Fix https://github.com/twbs/bootstrap/issues/15354

When you do this
image

You get this
image

### CODE REVIEW

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: b13efdea578cf6d6d2db331890a8ffdba2f684b9
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/50697412

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by vsn4ik** [comment]
`/^(show|hide)$/.test(option)` is shorter.

**Comment by Nikita240** [comment]
@vsn4ik I don't understand that.

**Comment by vsn4ik** [comment]
```js
if (option == 'show' || option == 'hide') { ... }
```
equivalent
```js
if (/^(show|hide)$/.test(option)) { ... }
```

**Comment by kkirsche** [comment]
only question I would ask @vsn4ik is while that is shorter I question if it sacrifices readability for such a small gain.

**Comment by vsn4ik** [comment]
Indeed. :+1: for the long version.

**Comment by Nikita240** [comment]
@vsn4ik Holy crap what syntax is that?

**Comment by kkirsche** [comment]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test explains it. It's just a regular expression tester

Please excuse brevity — Sent from my mobile device.

**Comment by Nikita240** [comment]
Ooooooh it's regex. That makes a lot more sense.

**Comment by cvrebert** [comment]
@Nikita240 Can you remove the `.min.*` files from this PR? They cause unnecessary merge conflicts.

**Comment by Nikita240** [comment]
@cvrebert Did I do it right?

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: d9981ae19a2b82d7262779af2b7ccbcf8bcbb4a8
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/51680981

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by cvrebert** [comment]
Not quite, but it's not a big deal. Don't worry about it.

CC: @fat for review

File: js/tests/unit/collapse.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ $(function () {
   })
 
   test('should hide a collapsed element', function () {
-    var $el = $('<div class="collapse"/>').bootstrapCollapse('hide')
+    var $el = $('<div class="collapse in"/>').bootstrapCollapse('hide')
```
Comment (cvrebert) [inline js/tests/unit/collapse.js]: A Collapsible with `.collapse.in` is in the shown/expanded (i.e. non-collapsed) state, so this change is invalid per the name of the test, AFAICT.

Comment (Nikita240) [inline js/tests/unit/collapse.js]: You're right. I just have to rename the test.

**Comment by cvrebert** [comment]
Superseded by #16011.

**Pull request closed without merging**